src/tito/release/distgit.py

```diff
@@ -424,6 +424,119 @@ def __init__(self, name=None, tag=None, build_dir=None,
             self.cli_tool = "rhpkg"
 
 
+class CentosGitReleaser(FedoraGitReleaser):
+    """
+    This process follows the choreography defined for the Centos9 Stream.
+    It will create a branch on the user's fork of the main repo.
+    That branch is merged to the main repo via pull request on the
+    web interface. There is no build option usable from this process
+    because it is not synchronous.
+
+    It is limited to only acting on a single Centos stream. Only the first
+    entry in 'branches' in the conf file will be used.
+    """
+
+    def __init__(self, name=None, tag=None, build_dir=None,
+            config=None, user_config=None,
+            target=None, releaser_config=None, no_cleanup=False,
+            test=False, auto_accept=False, **kwargs):
+        FedoraGitReleaser.__init__(self, name, tag, build_dir, config,
+                user_config, target, releaser_config, no_cleanup, test,
+                auto_accept, **kwargs)
+        # Override the cli_tool config from Fedora:
+        if "CENTPKG_USER" in user_config:
+            self.cli_tool = "centpkg --user={user}".format(user=user_config["CENTPKG_USER"])
+            self.username = user_config["CENTPKG_USER"]
+        else:
+            self.cli_tool = "centpkg"
+            self.username = getpass.getuser()
+
+    def _git_release(self):
+        getoutput("mkdir -p {working_dir}".format(working_dir=self.working_dir))
+        with chdir(self.working_dir):
+            run_command("{cli_tool} clone {project_name}".format(cli_tool=self.cli_tool,
+                                                                 project_name=self.project_name))
+
+        with chdir(self.package_workdir):
+            run_command("{cli_tool} fork".format(cli_tool=self.cli_tool))
+            run_command("git fetch {username}".format(username=self.username))
+            self.new_branch_name = "release-branch-{timestamp}".format(
+                timestamp=int(datetime.datetime.utcnow().timestamp()))
+            run_command("git checkout -b {new_branch_name} {username}/{branch}".format(
+                new_branch_name=self.new_branch_name,
+                username=self.username,
+                branch=self.git_branches[0]))
+
+        # Mead builds need to be in the git_root.  Other builders are agnostic.
+        with chdir(self.git_root):
+            self.builder.tgz()
+            self.builder.copy_extra_sources()
+
+        if self.test:
+            self.builder._setup_test_specfile()
+
+        self._git_sync_files(self.package_workdir)
+        self._git_upload_sources(self.package_workdir)
+        self._git_user_confirm_commit(self.package_workdir)
+
+    def _git_user_confirm_commit(self, project_checkout):
```

Comment:
It seems like this method is a copy-paste of `FedoraGitReleaser` implementation of this method except for one block that goes

```
for branch in self.git_branches[1:]:
    ...
```

Do you think it would be possible to drop this whole method and reuse the one from the parent? I can think of multiple ways to do that but some of them might not work, I didn't test it

1. Overriding the method and saving `self.git_branches` value in a temporary variable, setting `self.git_branches = self.git_branches[:1]` running `_git_user_confirm_commit` from parent and restoring the original `self.git_branches` value
2. Moving that one code block into a separate method (e.g. `_push_branches`) and then just calling it from `FedoraGitReleaser._git_user_confirm_commit`. Then overriding it here in `CentosGitReleaser` with just `pass`.
3. Possibly something else you can think of?

I know these are all workarounds but I think it would be better than having two implementations of such a long and complicated method.

I can also see that you are doing a custom `push` command so if we chose 2., we can also move that call to a `_push(self)` method, that you could override.